File: src/Roguelike/Optimization/Common/SimulationStats.cs
using System.Collections.Generic;

namespace Roguelike.Optimization
{


    public class SimulationStats
    {

        public bool IsVictory { get; set; }
        public int FinalFloorReached { get; set; }
        public float FinalHPPercent { get; set; }


        public List<string> MasterDeckIds { get; set; } = new List<string>();
        public List<string> RelicIds { get; set; } = new List<string>();


        public int ElitesDefeated { get; set; }
        public int ElitesEncountered { get; set; }
        public float TotalDamageTakenAtElites { get; set; }


        public int GoldCollected { get; set; }
        public int GoldSpent { get; set; }


        public Dictionary<string, int> CardPlayCounts { get; set; } = new Dictionary<string, int>();

        public SimulationStats() { }
    }
}

File: python_roguelike/optimization/simulation_stats.py
"""
simulation_stats.py  –  Python port of SimulationStats.cs

Records raw telemetry from a single game simulation run.
"""

from dataclasses import dataclass, field
from typing import Dict, List


@dataclass
class SimulationStats:
    is_victory: bool = False
    final_floor_reached: int = 0
    final_hp_percent: float = 0.0

    master_deck_ids: List[str] = field(default_factory=list)
    relic_ids: List[str] = field(default_factory=list)

    elites_defeated: int = 0
    elites_encountered: int = 0
    total_damage_taken_at_elites: float = 0.0

    gold_collected: int = 0
    gold_spent: int = 0

    card_play_counts: Dict[str, int] = field(default_factory=dict)
